src/common/openfam_wrappers.cpp

```diff
@@ -0,0 +1,1836 @@
+}
+
+const void* c_fam_get_option(c_fam* fam_obj, const char* optionName) {
+    const void *option = nullptr;
+    fam* fam_inst = (fam*)fam_obj;
+    try {
+        option = fam_inst->fam_get_option((char*)optionName);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return option;
+    }
+    return option;
+}
+
+const void* c_fam_get_option_ctx(c_fam* fam_obj, const char* optionName) {
+    const void *option = nullptr;
+    fam_context* fam_inst = (fam_context*)fam_obj;
+    try {
+        option = fam_inst->fam_get_option((char*)optionName);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return option;
+    }
+    return option;
+}
+
+int c_fam_stat_data(c_fam* fam_obj, c_fam_desc* fd, c_fam_stat* stat) {
+    fam* fam_inst = (fam*)fam_obj;
+    try {
+        fam_inst->fam_stat((Fd*)fd, (Fam_Stat*) stat);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return -1;
+    }
+    return 0;
+}
+
+int c_fam_stat_region(c_fam* fam_obj, c_fam_region_desc* region_desc, c_fam_stat* stat) {
+    fam* fam_inst = (fam*)fam_obj;
+    try {
+        fam_inst->fam_stat((Region*)region_desc, (Fam_Stat*)stat);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return -1;
+    }
+    return 0;
+}
+
+int c_fam_resize_region(c_fam* fam_obj, c_fam_region_desc* region_desc, uint64_t size) {
+    fam* fam_inst = (fam*)fam_obj;
+    try {
+        fam_inst->fam_resize_region((Region*)region_desc, size);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return -1;
+    }
+    return 0;
+}
+
+int c_fam_resize_region_ctx(c_fam* fam_obj, c_fam_region_desc* region_desc, uint64_t size) {
+    fam_context* fam_inst = (fam_context*)fam_obj;
+    try {
+        fam_inst->fam_resize_region((Region*)region_desc, size);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return -1;
+    }
+    return 0;
+}
+
+int c_fam_change_permissions_data(c_fam* fam_obj, c_fam_desc* fd, mode_t perm) {
+    fam* fam_inst = (fam*)fam_obj;
+    try {
+        fam_inst->fam_change_permissions((Fd*)fd, perm);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return -1;
+    }
+    return 0;
+}
+
+int c_fam_change_permissions_data_ctx(c_fam* fam_obj, c_fam_desc* fd, mode_t perm) {
+    fam_context* fam_inst = (fam_context*)fam_obj;
+    try {
+        fam_inst->fam_change_permissions((Fd*)fd, perm);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return -1;
+    }
+    return 0;
+}
+
+int c_fam_change_permissions_region(c_fam* fam_obj, c_fam_region_desc* region_desc, mode_t perm) {
+    fam* fam_inst = (fam*)fam_obj;
+    try {
+        fam_inst->fam_change_permissions((Region*)region_desc, perm);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return -1;
+    }
+    return 0;
+}
+
+// This implementation is currently mission in OpenFAM, hence keeping it commented
```

Comment:
The c++ API implementation is currently missing*

Reply:
Updated the comment.